src/macros.rs

```diff
@@ -178,6 +178,37 @@ macro_rules! unknown_define_index_type_option {
     () => {};
 }
 
+#[cfg(feature = "serde")]
+#[macro_export]
+#[doc(hidden)]
+macro_rules! maybe_index_impl_serde {
```

Comment:
nit: rename to `__internal_maybe_index_impl_serde` (here and below)

Reply:
Done.